### DISCUSSION TITLE
Variables/Metadata in YAML - specify a directory as a function of title

### DISCUSSION THREAD

**Discussion by TS-CUBED** [quarto-dev/quarto-cli#7990] [Q&A]
### Description

Hello, 

I'm trying to consolidate my YAML headers for my teaching material and can take most of it out of the qmd file. 

There are two settings that requires me to set a directory path to the `<filename>_files` directory.

Is there a way to specify this using a $( ), or {{ }} construct like one can in the code blocks or the text, but in the YAML header?

Alternative: is there a way to specify a relative path for these things, which starts from the one that is created on HTML output?

**Comment by mcanouil** [comment]
See <https://quarto.org/docs/authoring/variables.html>.

**Reply by TS-CUBED** [reply]
I found that page, and that works for using the title in the text, but doesn't seem to work for me when I use it in the YAML header. 

I need to use the variable (and ideally the filename, rather than the title) to construct the `<filename>_files` directory that Quarto uses for the external material in HTML output.

E.g., a file called `01-Introduction` would get a YAML header entry like:

```
      src: 01-Introduction_files/chalkboard.json
```

which I'd like to replace with something like:


```
      src: {{ filename }}_files/chalkboard.json
```

**Reply by cscheid** [reply]
This looks like a revealjs feature to re-use the chalkboard notes.

If I'm right, then I think you'd be better served by a javascript snippet that does that dynamic resolution at page-load time, which can be enabled by a Lua snippet that checks the YAML metadata for a flag (something like `use-saved-chalkboard: true` or whatever you prefer).
